rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -427,12 +423,12 @@ else if (isCached.get(idx).getLocation() == Location.CACHED) {
           if (CacheConfig.isParallelWarmupEnabled(conf)) {
             log.debug(String.format("Sending block %d to remoteFetchRequestChain", blockNum));
             if (directReadRequestChain == null) {
-              directReadRequestChain = new DirectReadRequestChain(getParentDataInputStream());
+              directReadRequestChain = ReadRequestChainFactory.createReadRequestChain(DirectReadRequestChain.class, getParentDataInputStream());
             }
 
             if (remoteFetchRequestChain == null) {
-              remoteFetchRequestChain = new RemoteFetchRequestChain(remotePath, remoteFileSystem,
-                  "localhost", conf, lastModified, fileSize, clusterType.ordinal());
+              remoteFetchRequestChain = ReadRequestChainFactory.createReadRequestChain(RemoteFetchRequestChain.class, "localhost", fileSize, lastModified, conf,
+                      remoteFileSystem, remotePath, clusterType.ordinal(), false, null, 0, 0);
```

Comment:
create a new method ReadRequestChainFactory.createReadRequestChain to work with  (RemoteFetchRequestChain.class, "localhost", fileSize, lastModified, conf,remoteFileSystem, remotePath, clusterType.ordinal()) and make it to call the existing createReadRequestChain by passing , false, null, 0, 0) additionally

Reply:
Done